File: anb/leetcode/binary/338. Counting Bits.py
class Solution(object):
    def countBits(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        ans = []
        ans.append(0)
        
        for i in range(1, n+1):
            for j in range(25, -1, -1):
                if (i & (1 << j)) > 0:
                    ans.append(ans[i ^ (1 << j)] + 1)
                    break
        return ans

File: jungclaire/leetcode/binary/338. Counting Bits.java
class Solution {
    public int[] countBits(int n) {
        int[] dp = new int[n + 1];

        for (int i = 1; i < dp.length; i++) {
            dp[i] = dp[i & (i - 1)] + 1;
        }

        return dp;
    }
}